utils/python/CIME/XML/machines.py

```diff
@@ -235,3 +235,33 @@ def get_batch_system_type(self):
         """
         batch_system = self.get_node("batch_system")
         return batch_system[0].get("type")
+
```

Comment:
Looks fine - the init_paths and and cmd_paths might have an optional lang argument?